File: LeetCode_1829/Main.py
class Solution(object):
    def getMaximumXor(self, nums, maximumBit):
        """
        :type nums: List[int]
        :type maximumBit: int
        :rtype: List[int]
        """
        mx = (1 << maximumBit) - 1
        n = len(nums)
        pre_xor = [0] * n
        ans = [0] * n
        
        pre_xor[0] = nums[0]
        for i in range(1, n):
            pre_xor[i] = pre_xor[i - 1] ^ nums[i]
        
        for i in range(n):
            ans[i] = pre_xor[n - i - 1] ^ mx
        
        return ans

File: LeetCode_1829/Main.java
public class Main {

    public static void main(String[] args) {
        
    }
}

class Solution {
    public int[] getMaximumXor(int[] nums, int maximumBit) {
        int mx = (1 << maximumBit) - 1;
        int n = nums.length;
        int[] ans = new int[n];
        int[] pre_xor = new int[n];
        
        pre_xor[0] = nums[0];
        for (int i = 1; i < n; i++) {
            pre_xor[i] = pre_xor[i - 1] ^ nums[i];
        }

        for (int i = 0; i < n; i++) {
            ans[i] = pre_xor[n - i - 1] ^ mx;
        }
        return ans;
    }
}
